sycl/include/CL/sycl/detail/pi.def

```diff
@@ -33,6 +33,7 @@ _PI_API(piextDeviceSelectBinary)
 _PI_API(piextGetDeviceFunctionPointer)
 _PI_API(piextDeviceGetNativeHandle)
 _PI_API(piextDeviceCreateWithNativeHandle)
+_PI_API(piextP2P)
```

Comment:
@smaslov-intel I think that note on Line 17 is not enough. We probably want some test, that would fail in a hard way.

Reply:
Yes, I missed the note.  I can apply its guidance in a new commit.

Reply:
The note is also a little ambiguous for me,  should I place new items at the end of the list or at the end of the category to which they fit. I.e. is piextP2P in the right place currently or should it be placed after _PI_API(piTearDown)?  Maybe I only have to move the Peer memcpy API's?

Reply:
@JackAKirk all new values should go after piTearDown. The idea is that those macros are expanded into a large enum, which is used in multiple places, including instrumentation APIs, that use values of that enum as keys to what PI call is being executed.

Reply:
OK makes sense thanks.

Reply:
`piextP2P` is not a very informative name. Maybe something like `piextDevicesSupportP2P` would be more telling?

Reply:
Sounds good. I made the suggested change.